boa/verifiers.py

```diff
@@ -0,0 +1,89 @@
+import json
+import time
+from datetime import datetime, timedelta
+from http import HTTPStatus
+from typing import Optional
+
+import requests
+from attr import dataclass
+
+from boa.util.abi import Address
+
+DEFAULT_BLOCKSCOUT_URI = "https://eth.blockscout.com"
+
+
+@dataclass
+class Blockscout:
+    """
+    Allows users to verify contracts on Blockscout.
+    This is independent of Vyper contracts, and can be used to verify any smart contract.
+    """
+
+    uri: str = DEFAULT_BLOCKSCOUT_URI
+    api_key: Optional[str] = None
+    timeout: timedelta = timedelta(minutes=2)
+    backoff: timedelta = timedelta(milliseconds=500)
+    backoff_factor: float = 1.1
+    retry_http_codes: tuple[int, ...] = (
+        HTTPStatus.NOT_FOUND,
+        HTTPStatus.INTERNAL_SERVER_ERROR,
```

Comment:
oh i didn't literally mean that we should handle these errors, i meant more that there could be unforeseeable errors that users will run into and they will end up needing to work around

Reply:
Well this allows them to override the statuses

Reply:
yea hm. what if we returned an awaitable which waits for the verification to complete? like typically the user does not actually need to wait for the verification to complete, although it is nice to give them the ability to loop until it does